operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -432,6 +449,66 @@ void reconcileIngressControllers() {
         }
     }
 
+    public void ensureBlueprintRouteMatching(Route route, String blueprintBaseName) {
+        var annotations = Optional.ofNullable(route.getMetadata()).map(ObjectMeta::getAnnotations);
+        var tslConfig = Optional.ofNullable(route.getSpec()).map(RouteSpec::getTls);
+
+        ensureBlueprintRouteMatching(annotations, tslConfig, blueprintBaseName);
+    }
+
+    public void ensureBlueprintRouteMatching(Optional<Map<String, String>> annotations, Optional<TLSConfig> tlsConfig, String blueprintBaseName) {
```

Comment:
`Optional<Map<K,V>>` seems an odd choice unless there is a semantic difference between no map and an empty one I'd prefer just to use `Map` here (though I'd rather ship than delay for polishing it can be polished later)

Reply:
I did that as the fabric8 API gives us the possibility of null annotation map on kube objects.